### PR TITLE
Navigation bar

### PR DESCRIPTION
**Pull request by QuicheLorraine3600**
According to the Figma, I added a navigation bar to our screens. The navigation bar is visible on all screens, except `LOADING_LOGIN`, `LOGIN_SCREEN` and `ACCOUNT_CREATION_SCREEN`. The tests have been updated accordingly.
I refactored `PrompLogin` to `promptLogin` to be able to call it from a non composable scope in order to send a guest user from the my events screen to the login screen.

Here is a short video showing the changes:

https://github.com/Chimpagne/ChimpagneApp/assets/39198541/6991b08c-1f91-4c85-a728-66475d3782a0

### CODE REVIEW

File: app/src/main/java/com/monkeyteam/chimpagne/ui/HomeScreen.kt [hunk 1]
```diff
@@ -206,115 +200,58 @@ fun HomeScreen(
...
                     }
                   },
                   modifier = Modifier.testTag("account_settings_button"))
             })
+      },
+      floatingActionButton = {
+        ExtendedFloatingActionButton(
+            onClick = { navObject.navigateTo(Route.EVENT_CREATION_SCREEN) },
+            icon = { Icon(Icons.Filled.Add, "organize an event") },
+            text = { Text(stringResource(R.string.homescreen_organize_event)) },
```
Comment (juaniac) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/HomeScreen.kt:215]: I personally think floating action buttons should not have text on them, plus "organize an event" takes way to much screen space (which makes it ugly):
image
So I am personally for the removal of this text field on this floating action button, just like for how you did on the figma:
image

**Review by juaniac — CHANGES_REQUESTED** [review]
Amazing PR overall.
This is a great refactoring of the main screen, which makes it in line with how many apps look like in the real world.
The code is very readable and from testing on the app seems robust and works as intended.
The modification of older files to work with the new navigation bar and tests were well done as well.

But there were a few oversights made which will need to be addressed to make this PR perfect. It shouldn't take more than 5 minutes to do all of my proposed requested changes.

**Review by juaniac — APPROVED** [review]
Requested changes were made,
Thus I approve this PR.

**Review by arnaudlvq — APPROVED** [review]
this really helps the user to navigate faster

**Pull request merged**